### DISCUSSION TITLE
creating single page pdf from multiple text files

### DISCUSSION THREAD

**Discussion by TheOriginalWr0ngway** [ImageMagick/ImageMagick#3434] [General]
I have four text files that I have broken out of a single 8 1/2 X 11 page that I'm trying to convert to a PDF file, using different point sizes for each file.

convert -append -compress zip -density 400 -font DejaVu-Sans-Mono-Book -pointsize 24 top.txt -pointsize 12 header.txt -pointsize 7 body.txt -pointsize 12 trailer.txt final.pdf

Problem is that I can't seem to get them all on the same page the append puts them all on the same page but It's 4 times the size of a single 8 1/2 X 11 page.   If I remove the -append then it's it's a 4 page PDF File.

Anyone know how to accomplish this?  It's really the first time I'm using convert and the doc pages are a little overwhelming.

Any help would be appreciated.
Steven

**Comment by snibgo** [comment]
What version of IM, on what platform?

If you use IM v7, I suggest using "magick", not "convert" or "magick convert".

Then put your operations in a logical order. "-append" must come after the input images, not before them.

You realise, I assume, that IM is a raster image processor. Your output PDF will contain a raster image, not searchable text.

As a final operation, you can use "-density" to get whatever page size (in inches) that you want.

**Reply by TheOriginalWr0ngway** [reply]
ŠOO[steven@pick tt]$ convert -V
Version: ImageMagick 6.7.2-7 2017-03-22 Q16 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2011 ImageMagick Studio LLC

Running on CentOS 6.10 (yes I know it EOL but it's still functional)

Using the -append after each input file still creates a pdf file with 4 letter size pdfs concatinated together.   The text file only contain the lines that need to be converted, 6, 6, 19 and 5 lines of text respectively.

**Comment by snibgo** [comment]
Your version of IM is very very old. I suggest you upgrade it. For recent IM, you need to prefix each input file with "TEXT:", eg "TEXT:top.txt".

You can do a single "-append" at the end, instead of after each input, if you want.

I suggest you try first with a single input file. After reading the text file, you can use "-trim" to remove the white border. Then you can use "-bordercolor white -border 5" to add a white border, if you want.

Operations will affect all images in the current list. Use parentheses to apply to just one image, like this:
```
convert \
  ( TEXT:top.txt -trim -bordercolor white -border 5 ) \
  ( TEXT:header.txt -trim -bordercolor white -border 5 ) \
  ( TEXT:bory.txt -trim -bordercolor white -border 5 ) \
  -append +repage \
  out.pdf
```

**Reply by TheOriginalWr0ngway** [reply]
Snigbo,
Thanks I was able to upgrade to Version: ImageMagick 7.0.11-5 Q16 x86_64 2021-03-20 https://imagemagick.org and I was finall able to get something close to what I need.   just need to figure out the switch to center one of the text files in the pdf.

I also notice that it didn't like parentheses and bash complained  about it being a syntax error.

I'm also using magick instead of convert Like you recommended.

But at least I have a new starting point to play with it
Thanks
Steven

**Reply by fmw42** [reply]
Syntax on Linux/Mac vs Windows is different. Unix syntax. (Snibgo forgot the \ escapes on the parentheses)

```
magick  \
 \ ( TEXT:top.txt -trim -bordercolor white -border 5 \) \
  \( TEXT:header.txt -trim -bordercolor white -border 5 \) \
  \( TEXT:bory.txt -trim -bordercolor white -border 5 \) \
  -append +repage \
  out.pdf

```

If you want centering, add -gravity center before -append. Or if you want it only horizontally centered use -gravity north.

**Reply by TheOriginalWr0ngway** [reply]
I also found that "-define trim:edges =north,south -trim" left the image centered, but I will definitely try your suggestion
